contrib/gate/entrypoint.sh

```diff
@@ -1,15 +1,23 @@
 #!/bin/bash
 
-[[ -n "$SRCPATH" ]] || \
-    ( echo "ERROR: \$SRCPATH must be non-empty" && exit 1 )
-[[ -n "$GOSRC" ]] || \
-    ( echo "ERROR: \$GOSRC must be non-empty" && exit 2 )
+set -e
+
+die() {
+    echo "${2:-FATAL ERROR (but no message given!) in ${FUNCNAME[1]}()}"
```

Comment:
A nice habit to develop is embedding a helpful prefix in `die` messages: something that explains where the message is coming from. Usually `$(basename $0)` is good, but in this case `entrypoint.sh` is useless so it should be something like `gate` or `gate-container` or something that will help a poor user track down where the error is actually coming from.

Reply:
Great suggestion.